tests/unit/test_download.py

```diff
@@ -676,16 +676,17 @@ def test_add_trusted_host__logging(self, caplog):
         assert actual == expected
 
     def test_iter_secure_origins(self):
```

Comment:
I think you should also add another test of `add_trusted_host()` that tests adding a hostname with a port.

Reply:
I think that is covered in `test_iter_secure_origins()`

Reply:
I would also add a couple test cases to `test_is_secure_origin` confirming that including a port has the desired behavior.

Reply:
I think it's good to test the lower-level functions in isolation as well, not just the higher-level.